core/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -231,6 +288,195 @@ class DeltaAnalysis(session: SparkSession)
+        }
+        // We're creating a table by path and there won't be a place to store catalog stats
+        val catalog = createCatalogForCommandForClone(
+          path, byPath = true, tblIdent, targetLocation, sourceCatalogTable, sourceTbl)
+        CreateDeltaTableCommand(
+          catalog,
+          None,
+          saveMode,
+          Some(CloneTableCommand(
+            sourceTbl,
+            tblIdent,
+            statement.tablePropertyOverrides,
+            path)),
+          tableByPath = true,
+          output = CloneTableCommand.output)
+
+      // Target is a metastore table
+      case UnresolvedRelation(SessionCatalogAndIdentifier(catalog, ident), _, _) =>
+        if (!isCreate) {
+          throw DeltaErrors.cannotReplaceMissingTableException(ident)
+        }
+        val tblIdent = ident
+          .asTableIdentifier
+        val finalTarget = new Path(statement.targetLocation.getOrElse(
+          session.sessionState.catalog.defaultTablePath(tblIdent).toString))
+        val catalogTable = createCatalogForCommandForClone(
+          finalTarget, byPath = false, tblIdent, targetLocation, sourceCatalogTable, sourceTbl)
+        val catalogTableWithPath = if (targetLocation.isEmpty) {
+          catalogTable.copy(
+            storage = CatalogStorageFormat.empty.copy(locationUri = Some(finalTarget.toUri)))
+        } else {
+          catalogTable
+        }
+        CreateDeltaTableCommand(
+          catalogTableWithPath,
+          None,
+          saveMode,
+          Some(CloneTableCommand(
+            sourceTbl,
+            tblIdent,
+            statement.tablePropertyOverrides,
+            finalTarget)),
+          operation = tableCreationMode,
+          output = CloneTableCommand.output)
+
+      // Delta metastore table already exists at target
+      case DataSourceV2Relation(
+          deltaTableV2 @ DeltaTableV2(_, path, existingTable, _, _, _, _), _, _, _, _) =>
+        val tblIdent = existingTable match {
+          case Some(existingCatalog) => existingCatalog.identifier
+          case None => TableIdentifier(path.toString, Some("delta"))
+        }
+        val cloneSourceTable = sourceTbl
+
+
+        val catalog = createCatalogForCommandForClone(
+          path,
+          byPath = existingTable.isEmpty,
+          tblIdent,
+          targetLocation,
+          sourceCatalogTable,
+          cloneSourceTable)
+
+        CreateDeltaTableCommand(
+          catalog,
+          existingTable,
+          saveMode,
+          Some(CloneTableCommand(
+            cloneSourceTable,
+            tblIdent,
+            statement.tablePropertyOverrides,
+            path)),
+          tableByPath = existingTable.isEmpty,
+          operation = tableCreationMode,
+          output = CloneTableCommand.output)
+
+      // Non-delta metastore table already exists at target
+      case LogicalRelation(_, _, existingCatalog @ Some(catalog), _) =>
```

Comment:
What do we do if there is a non-delta table exists in catalog which is same as the clone target table?

Reply:
It would depend on the SaveMode, e.g. [here](https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala#L70)